EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -1874,3 +2042,309 @@ std::string V2::World::clipCountryFileName(const std::string& incoming) const
 	else
 		return incoming.substr(0, 76) + ".txt";
 }
+
+
+void V2::World::copyFolder(const std::string& folder, const std::string& src, const std::string& dest) const
+{
+	if (commonItems::DoesFolderExist(src + "/" + folder))
+	{
+		fs::copy(src + "/" + folder, dest + "/" + folder, fs::copy_options::recursive);
+	}
+}
+
+void V2::World::mergeFolder(const std::string& folder, const std::string& src, const std::string& dest) const
+{
+	const auto& files = commonItems::GetAllFilesInFolder(src + "/" + folder);
+	for (const auto& file: files)
+	{
+		fs::copy_file(src + "/" + folder + "/" + file, dest + "/" + folder + "/" + file, fs::copy_options::recursive);
+	}
+}
+
+void V2::World::copyModFiles() const
+{
+	if (!theConfiguration.getVic2ModName().empty())
+	{
+		LOG(LogLevel::Info) << "<- Copying " + theConfiguration.getVic2ModName() + " files";
+		const auto& mod = theConfiguration.getVic2ModPath() + "/" + theConfiguration.getVic2ModName();
+		const auto& configurables = "configurables/" + theConfiguration.getVic2ModName();
+		const auto& output = "output/" + theConfiguration.getOutputName();
+
+		const auto& commons = commonItems::GetAllFilesInFolder(mod + "/common");
+		for (const auto& file: commons)
+		{
+			if (file == "rebel_types.txt")
+			{
+				fs::remove(output + "/common/rebel_types.txt");
+			}
+			else if (commonItems::DoesFileExist(output + "/common/" + file))
+			{
+				continue;
+			}
+			fs::copy_file(mod + "/common/" + file, output + "/common/" + file);
+		}
+
+		if (commonItems::DoesFileExist(configurables + "/religion.txt"))
+		{
+			fs::remove(output + "/common/religion.txt");
+			fs::copy_file(configurables + "/religion.txt", output + "/common/religion.txt");
+		}
+
+		copyFolder("battleplans", mod, output);
+		copyFolder("map", mod, output);
+
+		mergeFolder("gfx/interface/leaders", mod, output);
+		const auto& gfxInterfaceFiles = commonItems::GetAllFilesInFolder(mod + "/gfx/interface");
+		for (const auto& file: gfxInterfaceFiles)
+		{
+			if (file != "icon_religion.dds")
```

Comment:
don't mess with religions. unified religion file, unified religion strips from our blankMod. HPM doesn't support a quarter of stuff we need. ditto for religion.txt above.